pkg/chains/storage/docdb/docdb.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docdb
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"github.com/tektoncd/chains/pkg/config"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	"gocloud.dev/docstore"
+	_ "gocloud.dev/docstore/gcpfirestore"
+)
+
+const (
+	StorageTypeDocDB = "docdb"
+)
+
+// Backend is a storage backend that stores signed payloads in the TaskRun metadata as an annotation.
+// It is stored as base64 encoded JSON.
+type Backend struct {
+	logger *zap.SugaredLogger
+	tr     *v1beta1.TaskRun
+	coll   *docstore.Collection
+}
+
+type SignedDocument struct {
+	Signed    []byte
+	Signature string
+	Object    interface{}
+	Name      string
+}
+
+// NewStorageBackend returns a new Tekton StorageBackend that stores signatures on a TaskRun
+func NewStorageBackend(logger *zap.SugaredLogger, tr *v1beta1.TaskRun, cfg config.Config) (*Backend, error) {
+	url := cfg.Storage.DocDB.URL
+	coll, err := docstore.OpenCollection(context.Background(), url)
+	if err != nil {
+		return nil, err
+	}
+
+	return newStorageBackendWithColl(logger, tr, coll), nil
+}
+
+func newStorageBackendWithColl(logger *zap.SugaredLogger, tr *v1beta1.TaskRun, coll *docstore.Collection) *Backend {
+	return &Backend{
+		logger: logger,
+		tr:     tr,
+		coll:   coll,
+	}
+}
+
+// StorePayload implements the Payloader interface.
+func (b *Backend) StorePayload(signed []byte, signature string, key string) error {
+	// We need two object names: the signature and the payload. We want to make these unique to the UID, but easy to find based on the
+	// name/namespace as well.
+	// $bucket/taskrun-$namespace-$name-$uid/$key.signature
+	// $bucket/taskrun-$namespace-$name-$uid/$key.payload
+	root := fmt.Sprintf("taskrun-%s-%s-%s", b.tr.Namespace, b.tr.Name, b.tr.UID)
+	sigName := path.Join(root, fmt.Sprintf("%s.signature", key))
```

Comment:
`sigName` is never used except for the log message. In the signed document, `Name` is set to `key` (which is the last part of `sigName`), that's not where `sigName` was meant to go?

Reply:
Also two names are mentioned in the comment, but only one (`sigName`) is generated.

Reply:
Thanks! copy/paste mistake from the GCS storage one :)